src/crafting.h

```diff
@@ -69,6 +69,8 @@ struct recipe {
         bool hidden;
     };
     std::vector<bookdata_t> booksets;
+    std::set<std::string> flags;
```

Comment:
> Adding two bools or a float makes this too single-purpose.

The flags only have one purpose to start with. Why make it needlessly generic? If someone later needs generic flags (not named at compile time), they can implement it at that point.

> I anticipate adding more flags to recipes in the future.

Fine. But those flag most likely require changes to the code anyway, adding another member and the code to load it from JSON doesn't matter much.

> An array of strings is how flags are handled in every other object type we deal with.

The project used to store raw enums in the saves, that wasn't a good approach, but it was commonly used, so I'm leaning towards storing flags as set of strings being cargo cult. Besides: there are several instances where it's done differently: monster type flags use a bitset, which is essentially a vector of bools. Several other types have bool members that act as flags.

> he was "leaning toward a flag".

A member `bool flag;` *is* a flag.